### PR TITLE
UI: Fix layout height

### LINKED ISSUE #24261: Remove `height: 100svh` in favour of a solution that work for Chrome 100 and above

### PR DESCRIPTION
**Pull request by cdedreuille**
Closes https://github.com/storybookjs/storybook/issues/24261

## What I did

To support older browsers we can't use `100dvh` to set the height of the layout to be always visible on mobile. Instead, the most reliable way is to set the height using javascript. A common pattern is to set a css variable for `--vh` that you can then use elsewhere if needed it on your app.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: code/ui/manager/src/components/layout/Layout.tsx [hunk 1]
```diff
@@ -171,9 +171,22 @@ export const Layout = ({ managerLayoutState, setManagerLayoutState, ...slots }:
...
+    useEffect(() => {
+      setHeight();
+      window.addEventListener('resize', setHeight);
+      return () => window.removeEventListener('resize', setHeight);
+    }, []);
+
     return {
       width: '100%',
-      height: '100svh', // We are using svh to use the minimum space on mobile
+      height: 'var(--vh, 100vh)',
```
Comment (ndelangen) [inline code/ui/manager/src/components/layout/Layout.tsx:189]: Emotion supports array values for fallbacks:
https://emotion.sh/docs/object-styles#fallbacks

```suggestion
      height: ['100dvh', '100vh],
```

IIUC, the bug that requires `100dvh` only happens on `iOS` and that supports the CSS feature?

Comment (JReinhold) [reply code/ui/manager/src/components/layout/Layout.tsx:189]: This is more about supporting old browser versions, and we haven't yet decided which versions to support in SB 8.0. https://caniuse.com/viewport-unit-variants

The fallback syntax looks interesting, I can't find any info on how this works? Is it a browser feature, or does Emotion do something clever with a list of CSS features per user agent?

based on the docs you linked, I think the syntax is the other way around in your suggestion.

Comment (cdedreuille) [reply code/ui/manager/src/components/layout/Layout.tsx:189]: Oh that's interesting. I didn't know Emotion had fallbacks. I'll try it out now.

Comment (ndelangen) [reply code/ui/manager/src/components/layout/Layout.tsx:189]: Having the same property twice on a single class is a CSS feature.
CSS is fault-tolerant.

Emotion allows you to use this feature with the array syntax.

True, the exact implementation might have to be reversed from the code suggestion I added.

**Comment by cdedreuille** [comment]
@JReinhold @ndelangen I've implemented the change and removed the JS code 😊 It does work as intended.

CleanShot 2023-10-06 at 16 42 38@2x

**Review by ndelangen — APPROVED** [review]

**Review by ndelangen — APPROVED** [review]

**Pull request merged**